src/AWS.jl

```diff
@@ -57,6 +56,47 @@ end
 struct RestJSONService
     name::String
     api_version::String
+
+    service_specific_headers::LittleDict{String, String}
+
+    RestJSONService(name::String, api_version::String) = new(name, api_version, LittleDict())
+    RestJSONService(name::String, api_version::String, service_specific_headers::LittleDict{String, String}) = new(name, api_version, service_specific_headers)
+end
+
+Base.@kwdef mutable struct Request
+    service::String
+    api_version::String
+    request_method::String
+
+    headers::AbstractDict{String, String}=LittleDict{String, String}()
+    content::String=""
+    resource::String=""
+    url::String=""
+
+    return_stream::Bool=false
+    response_stream::Union{Base.BufferStream, Nothing}=nothing
+    http_options::Array{String}=[]
+    return_raw::Bool=false
+    response_dict_type::Type=LittleDict
```

Comment:
Doing this removes the need for the inner constructor I believe:
```suggestion
    response_dict_type::Type{<:AbstractDict}=LittleDict
```